src/graph/entities/qg_node.c

```diff
@@ -131,16 +175,30 @@ void QGNode_AddLabel
 (
 	QGNode *n,
 	const char *l,
-	int l_id
+	int l_id,
+	bool optional
 ) {
 	ASSERT(n != NULL);
 	ASSERT(l != NULL);
 
-	// node already labeled as l
-	if(QGNode_HasLabel(n, l)) return;
+	// Mandatory > Optional label (stronger) --> If n already has label 
+	// l, and it is non optional, and optional=false update n's optional entry
+	// corresponding to l.
```

Comment:
Please try to rephrase this comment.
`n->optional[i] = n->optional[i] | !optional`